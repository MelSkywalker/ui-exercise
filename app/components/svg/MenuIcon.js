import React from 'react';

const MenuIcon = () => {
	return (
		<svg className="svg menu-icon" viewBox="0 0 100 80" width="20" height="20">
			<rect width="80" height="10"></rect>
			<rect y="30" width="80" height="10"></rect>
			<rect y="60" width="80" height="10"></rect>
		</svg>
	);
}

export default MenuIcon;