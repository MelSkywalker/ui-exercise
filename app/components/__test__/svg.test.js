import React from 'react';
import { shallow } from 'enzyme';

import GmailIcon from '../svg/GmailIcon';
import MenuIcon from '../svg/MenuIcon';
import InboxIcon from '../svg/InboxIcon';
import ImportantIcon from '../svg/ImportantIcon';

describe('components | svg', () => {
	it('renders the GmailIcon component', () => {
		const component = shallow(<GmailIcon />);
		expect(component).toMatchSnapshot();
	});
	it('renders the MenuIcon component', () => {
		const component = shallow(<MenuIcon />);
		expect(component).toMatchSnapshot();
	});
	it('renders the InboxIcon component', () => {
		const component = shallow(<InboxIcon />);
		expect(component).toMatchSnapshot();
	});
	it('renders the ImportantIcon component', () => {
		const component = shallow(<ImportantIcon />);
		expect(component).toMatchSnapshot();
	});
});
