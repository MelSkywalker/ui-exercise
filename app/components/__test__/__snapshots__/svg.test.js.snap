// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | svg renders the GmailIcon component 1`] = `
<svg
  className="svg gmail-icon"
  height="2500"
  viewBox="0 49.4 512 399.4200000000001"
  width="2500"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
  >
    <g
      fillRule="nonzero"
    >
      <path
        d="M34.91 448.818h81.454V251L0 163.727V413.91c0 19.287 15.622 34.91 34.91 34.91z"
        fill="#4285f4"
      />
      <path
        d="M395.636 448.818h81.455c19.287 0 34.909-15.622 34.909-34.909V163.727L395.636 251z"
        fill="#34a853"
      />
      <path
        d="M395.636 99.727V251L512 163.727v-46.545c0-43.142-49.25-67.782-83.782-41.891z"
        fill="#fbbc04"
      />
    </g>
    <path
      d="M116.364 251V99.727L256 204.455 395.636 99.727V251L256 355.727z"
      fill="#ea4335"
    />
    <path
      d="M0 117.182v46.545L116.364 251V99.727L83.782 75.291C49.25 49.4 0 74.04 0 117.18z"
      fill="#c5221f"
      fillRule="nonzero"
    />
  </g>
</svg>
`;

exports[`components | svg renders the ImportantIcon component 1`] = `
<svg
  className="svg important-icon"
  height="14px"
  version="1.1"
  viewBox="0 0 17 14"
  width="17px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Icons"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Rounded"
      transform="translate(-716.000000, -335.000000)"
    >
      <g
        id="Action"
        transform="translate(100.000000, 100.000000)"
      >
        <g
          id="-Round-/-Action-/-label_important"
          transform="translate(612.000000, 230.000000)"
        >
          <g
            transform="translate(0.000000, 0.000000)"
          >
            <polygon
              id="Path"
              points="0 0 24 0 24 24 0 24"
            />
            <path
              d="M5.94,18.99 L15,18.99 C15.65,18.99 16.26,18.68 16.63,18.15 L20.58,12.58 C20.83,12.23 20.83,11.77 20.58,11.42 L16.62,5.84 C16.26,5.31 15.65,5 15,5 L5.94,5 C5.13,5 4.66,5.93 5.13,6.59 L9,12 L5.13,17.41 C4.66,18.07 5.13,18.99 5.94,18.99 Z"
              fill="#1D1D1D"
              id="ðŸ”¹Icon-Color"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`components | svg renders the InboxIcon component 1`] = `
<svg
  className="svg inbox-icon"
  height="18px"
  version="1.1"
  viewBox="0 0 18 18"
  width="18px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Icons"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Rounded"
      transform="translate(-817.000000, -1529.000000)"
    >
      <g
        id="Content"
        transform="translate(100.000000, 1428.000000)"
      >
        <g
          id="-Round-/-Content-/-move_to_inbox"
          transform="translate(714.000000, 98.000000)"
        >
          <g>
            <polygon
              id="Path"
              points="0 0 24 0 24 24 0 24"
            />
            <path
              d="M19,3 L4.99,3 C3.88,3 3,3.9 3,5 L3,19 C3,20.1 3.88,21 4.99,21 L19,21 C20.1,21 21,20.1 21,19 L21,5 C21,3.9 20.1,3 19,3 Z M19,15 L15.87,15 C15.4,15 15.02,15.34 14.89,15.8 C14.54,17.07 13.37,18 12,18 C10.63,18 9.46,17.07 9.11,15.8 C8.98,15.34 8.6,15 8.13,15 L4.99,15 L4.99,6 C4.99,5.45 5.44,5 5.99,5 L18,5 C18.55,5 19,5.45 19,6 L19,15 Z M16,10 L14,10 L14,7 L10,7 L10,10 L8,10 L11.65,13.65 C11.85,13.85 12.16,13.85 12.36,13.65 L16,10 Z"
              fill="#1D1D1D"
              id="ðŸ”¹Icon-Color"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;

exports[`components | svg renders the MenuIcon component 1`] = `
<svg
  className="svg menu-icon"
  height="20"
  viewBox="0 0 100 80"
  width="20"
>
  <rect
    height="10"
    width="80"
  />
  <rect
    height="10"
    width="80"
    y="30"
  />
  <rect
    height="10"
    width="80"
    y="60"
  />
</svg>
`;

exports[`components | svg renders the TrashIcon component 1`] = `
<svg
  className="svg trash-icon"
  height="18px"
  version="1.1"
  viewBox="0 0 14 18"
  width="14px"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="none"
    fillRule="evenodd"
    id="Icons"
    stroke="none"
    strokeWidth="1"
  >
    <g
      id="Rounded"
      transform="translate(-275.000000, -245.000000)"
    >
      <g
        id="Action"
        transform="translate(100.000000, 100.000000)"
      >
        <g
          id="-Round-/-Action-/-delete"
          transform="translate(170.000000, 142.000000)"
        >
          <g>
            <polygon
              id="Path"
              points="0 0 24 0 24 24 0 24"
            />
            <path
              d="M6,19 C6,20.1 6.9,21 8,21 L16,21 C17.1,21 18,20.1 18,19 L18,9 C18,7.9 17.1,7 16,7 L8,7 C6.9,7 6,7.9 6,9 L6,19 Z M18,4 L15.5,4 L14.79,3.29 C14.61,3.11 14.35,3 14.09,3 L9.91,3 C9.65,3 9.39,3.11 9.21,3.29 L8.5,4 L6,4 C5.45,4 5,4.45 5,5 C5,5.55 5.45,6 6,6 L18,6 C18.55,6 19,5.55 19,5 C19,4.45 18.55,4 18,4 Z"
              fill="#1D1D1D"
              id="ðŸ”¹Icon-Color"
            />
          </g>
        </g>
      </g>
    </g>
  </g>
</svg>
`;
